### PR TITLE
Add warning that `fork` for the `mp_context` is not recommended for macOS

### PR DESCRIPTION
**Pull request by zm711**
Actually I was reading up on multiprocessing and learned that as of 3.8 `fork` should really be avoided on macOS since the resource manager might spin threads. Docs [here](https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods). 

Although fork can be used on macOS it is not guaranteed to work safely ( and now `spawn` is the macOS and Windows default), so I added a warning into the job_tools to warn users that setting to fork on macOS is no longer considered safe.

I could change Linux to "not macOS POSIX-based systems" if you'd like though.

+ I noticed one typo in the workflow I wrote for `check_your_install` so I fixed that.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
The standard advice is that fork should not be used anywhere. They are changing the default in python 3.13. I have say before that we should change the default in general.

**Comment by h-mayorquin** [comment]
See here:
https://github.com/python/cpython/issues/84559

@alejoe91 @samuelgarcia should we change this? 

I know that fork is faster rand probably faster in some cases but the default should not be something that could be unsafe. We should make the people that want to take the risk opt-in instead of opt-out.

**Comment by h-mayorquin** [comment]
See also:
https://discuss.python.org/t/concerns-regarding-deprecation-of-fork-with-alive-threads/33555

**Comment by zm711** [comment]
We could definitely just add a general warning for all OSs instead, but if mac or linux users want to force it for now they still can. At least for mac the default context from multiprocessing has already switched. But I'm with @h-mayorquin, in that risky behavior should be opt-in instead of opt-out.

**Review by alejoe91 — APPROVED** [review]

**Comment by zm711** [comment]
So I fixed my elif which would give the warning even if the user gave 'spawn' for Mac. And added a general warning. I suppose we could just wait for python to switch the mp context itself, although maybe in a separate PR we could discuss switching the default to 'spawn' no matter what?

**Comment by alejoe91** [comment]
Guys, let's discuss more about adding the `warning` on linux in general. I would say to wait until the next release!

**Comment by zm711** [comment]
> Guys, let's discuss more about adding the warning on linux in general. I would say to wait until after the release!

Done

**Comment by samuelgarcia** [comment]
Just to clarify. We need to keep fork in linux. This is more efficient.
The warning is when we already have thread which is not our case at all. So it is safe.
We can optionally have thread starting in the forked process but our default is one thread.
So I think our fork is safe in linux.
I am OK for the darwin warning but not at all when linux.
I also think that even when python will remove "fork" as default we should maintain the "fork" as default on linux on our side as soon as it work because we don't have threads.
In short lets merge this since Zach removed it.

**Review by samuelgarcia — APPROVED** [review]

**Comment by h-mayorquin** [comment]
The intention was doing after release, but is not about threads Sam. Read the discussion by the python mantainers that I shared. Quote:

> While that is a great thing for you to do, unfortunately on POSIX doing enough isn’t actually possible. It is impossible to safely code threads that can be used in an application that calls fork() in Python **because the CPython runtime itself is by definition unsafe for use after fork().** 

https://discuss.python.org/t/concerns-regarding-deprecation-of-fork-with-alive-threads/33555


Se this for a recollection of problems.
https://github.com/python/cpython/issues/84559
